webapi/src/main/scala/org/knora/webapi/util/PermissionUtilV1.scala

```diff
@@ -401,11 +406,212 @@ object PermissionUtilV1 {
+
+                        permissionType match {
+                            case PermissionType.AP => {
+                                if (!OntologyConstants.KnoraBase.AdministrativePermissionAbbreviations.contains(abbreviation)) {
+                                    throw InconsistentTriplestoreDataException(s"Unrecognized permission abbreviation '$abbreviation'")
+                                }
+                                if (splitPermission.length > 1) {
+                                    val shortGroups: Array[String] = splitPermission(1).split(OntologyConstants.KnoraBase.GroupListDelimiter)
+                                    val groups: Set[IRI] = shortGroups.map(_.replace(OntologyConstants.KnoraBase.KnoraBasePrefix, OntologyConstants.KnoraBase.KnoraBasePrefixExpansion)).toSet
+                                    buildPermissionObject(abbreviation, groups)
+                                } else {
+                                    buildPermissionObject(abbreviation, Set.empty[IRI])
+                                }
+                            }
+                            case PermissionType.OAP => {
+                                if (!OntologyConstants.KnoraBase.ObjectAccessPermissionAbbreviations.contains(abbreviation)) {
+                                    throw InconsistentTriplestoreDataException(s"Unrecognized permission abbreviation '$abbreviation'")
+                                }
+                                val shortGroups: Array[String] = splitPermission(1).split(OntologyConstants.KnoraBase.GroupListDelimiter)
+                                val groups: Set[IRI] = shortGroups.map(_.replace(OntologyConstants.KnoraBase.KnoraBasePrefix, OntologyConstants.KnoraBase.KnoraBasePrefixExpansion)).toSet
+                                buildPermissionObject(abbreviation, groups)
+                            }
+                        }
+                }
+            }.toSet
+            case None => Set.empty[PermissionV1]
+        }
+    }
+
+    /**
+      * Helper method used to convert the permission string stored inside the triplestore to a permission object.
+      *
+      * @param name the name of the permission.
+      * @param iris the optional set of additional information (e.g., group IRIs, resource class IRIs).
+      * @return a sequence of permission objects.
+      */
+    def buildPermissionObject(name: String, iris: Set[IRI]): Set[PermissionV1] = {
+        name match {
+            case OntologyConstants.KnoraBase.ProjectResourceCreateAllPermission => Set(PermissionV1.ProjectResourceCreateAllPermission)
+            case OntologyConstants.KnoraBase.ProjectResourceCreateRestrictedPermission => {
+                if (iris.nonEmpty) {
+                    iris.map(iri => PermissionV1.ProjectResourceCreateRestrictedPermission(iri))
+                } else {
+                    throw InconsistentTriplestoreDataException(s"Missing additional permission information!")
```

Comment:
lol. keep calm :-) 

![](https://upload.wikimedia.org/wikipedia/commons/6/6f/Keep-calm-and-carry-on-scan.jpg)